pkg/identity/identitysync/okta.go

```diff
@@ -94,12 +97,16 @@ func (o *OktaSync) idpUserFromOktaUser(ctx context.Context, oktaUser *okta.User)
 
 	log.Debug("listing groups for user")
 
-	userGroups, _, err := o.client.User.ListUserGroups(ctx, oktaUser.Id)
-	if err != nil {
-		return u, errors.Wrapf(err, "listing groups for okta user %s", oktaUser.Id)
-	}
-	for _, g := range userGroups {
-		u.Groups = append(u.Groups, g.Id)
+	if userIdToGroupIds != nil {
```

Comment:
nit: for readbility,  instead of having this function behaving in 2 different ways, I would do the logic for the groups in each loop of each `listAllUsers()` and `listUsersByGroups()`